src/transform/src/use_indexes.rs

```diff
@@ -63,283 +75,60 @@ impl FilterEqualLiteral {
-                .zip(inputs.iter())
-                .enumerate()
-            {
-                if let RelationExpr::Filter { input, predicates } = join_input {
-                    // if none of the predicates refer to join keys, it can be lifted
-                    if predicates.iter().all(|p| {
-                        let support = p.support();
-                        !join_keys.iter().any(|k| support.contains(k))
-                    }) {
-                        add_matching_index_by_input(
-                            &**input,
-                            args.indexes,
-                            input_num,
-                            join_keys,
-                            &mut matching_index_by_input,
-                        );
-                    }
-                } else {
-                    add_matching_index_by_input(
-                        join_input,
-                        args.indexes,
-                        input_num,
-                        join_keys,
-                        &mut matching_index_by_input,
-                    );
-                }
-            }
-            if !matching_index_by_input.is_empty() {
-                // do a topological sort of the variables. There exists a directed edge from
-                // equivalence class x to class y if there exist a variable z in x and
-                // a variable b in y such that there exists an index where z and b are keys,
-                // and z comes right before b.
-                let mut incoming_edges = std::iter::repeat(vec![])
-                    .take(variables.len())
-                    .collect::<Vec<_>>();
-                for (input_num, index) in matching_index_by_input.iter() {
-                    for k in 1..index.len() {
-                        if let ScalarExpr::Column(c_from) = index[k - 1] {
-                            if let ScalarExpr::Column(c_to) = index[k] {
-                                let from_class = variables
-                                    .iter()
-                                    .position(|eq| eq.contains(&(*input_num, c_from)));
-                                let to_class = variables
-                                    .iter()
-                                    .position(|eq| eq.contains(&(*input_num, c_to)));
-                                if let Some(from_position) = from_class {
-                                    if let Some(to_position) = to_class {
-                                        if !incoming_edges[to_position].contains(&from_position) {
-                                            incoming_edges[to_position].push(from_position)
-                                        }
-                                    }
+                                (ScalarExpr::Column(_), ScalarExpr::Literal(_, _)) => {
+                                    expr_to_equivalent_literal
+                                        .insert((**expr2).clone(), (**expr1).clone());
```

Comment:
Should these be reversed? They seem to be the same as above, but here the role of column and literal are swapped.

Reply:
Yes they should be! Made a typo while trying to break out the code from the other PR. Thanks for catching this!